django/template/backends/base.py

```diff
@@ -23,6 +23,12 @@ def __init__(self, params):
                 "Unknown parameters: {}".format(", ".join(params))
             )
 
+    def check(self, **kwargs):
+        raise NotImplementedError(
+            "subclasses may provide a check() method to verify the template engine "
+            "is configured correctly."
+        )
```

Comment:
I know this is copied from the finders, but now I think about it, `NotImplementedError` is a bit weird. It forces the caller to catch it, complicating the implementation. We don’t do that for most other base class methods, like `Model.check()` or whatever.

Returning an empty list makes more sense to me:

```suggestion
        return []
```

Reply:
Yeah good point, this makes more sense to me too,